radicle-node/src/service/tracking/store.rs

```diff
@@ -0,0 +1,397 @@
+use std::path::Path;
+use std::str::FromStr;
+use std::{fmt, io};
+
+use sqlite as sql;
+use thiserror::Error;
+
+use crate::prelude::Id;
+use crate::service::NodeId;
+
+use super::{Policy, Scope};
+
+#[derive(Error, Debug)]
+pub enum Error {
+    /// I/O error.
+    #[error("i/o error: {0}")]
+    Io(#[from] io::Error),
+    /// An Internal error.
+    #[error("internal error: {0}")]
+    Internal(#[from] sql::Error),
+}
+
+pub type Alias = String;
+
+impl sqlite::BindableWithIndex for Scope {
+    fn bind<I: sql::ParameterIndex>(self, stmt: &mut sql::Statement<'_>, i: I) -> sql::Result<()> {
+        let s = match self {
+            Self::Trusted => "trusted",
+            Self::DelegatesOnly => "delegates-only",
+            Self::All => "all",
```

Comment:
would a 'to_string()' make sense here for the Scope type?

Reply:
Yeah, I think we will need that eventually, so will leave this as-is for now and when it's needed I'll replace this with to_string.